arci/src/clients/partial_joint_trajectory_client.rs

```diff
@@ -30,13 +30,33 @@ impl<C> PartialJointTrajectoryClient<C>
 where
     C: JointTrajectoryClient,
 {
-    pub fn new(joint_names: Vec<String>, shared_client: C) -> Self {
+    pub fn try_new(joint_names: Vec<String>, shared_client: C) -> Result<Self, String> {
         let full_joint_names = shared_client.joint_names().to_vec();
-        Self {
+        if joint_names.len() > full_joint_names.len() {
+            return Err(format!(
+                "Joint name length is illegal. (full = { }, partial = { })",
+                full_joint_names.len(),
+                joint_names.len()
+            ));
+        }
+
+        let mut partial = Vec::with_capacity(joint_names.len());
+        for joint_name in joint_names.iter() {
+            if let Some(_) = full_joint_names.iter().find(|&x| x == joint_name) {
```

Comment:
I think it's enough to return Err at the moment when any item of joint_names is not found in full_joint_names.

Reply:
I changed it.
Please, check.